misc/python/materialize/checks/all_checks/retain_history.py

```diff
@@ -0,0 +1,87 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.checks.executors import Executor
+from materialize.mz_version import MzVersion
+
+
+class RetainHistory(Check):
+    def _can_run(self, e: Executor) -> bool:
+        return e.current_mz_version >= MzVersion.parse_mz("v0.81.0")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                $ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+                ALTER SYSTEM SET enable_logical_compaction_window = true
+
+                > CREATE TABLE retain_history_table (key INT NOT NULL, value INT NOT NULL);
+                > INSERT INTO retain_history_table VALUES (1, 100), (2, 200);
+
+                > CREATE MATERIALIZED VIEW retain_history_mv1 WITH (RETAIN HISTORY FOR '30s') AS
+                    SELECT * FROM retain_history_table;
+
+                > SELECT count(*) FROM retain_history_mv1;
+                2
+            """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > UPDATE retain_history_table SET value = value + 10 WHERE key = 1;
+
+                > CREATE MATERIALIZED VIEW retain_history_mv2 WITH (RETAIN HISTORY FOR '30s') AS
+                    SELECT * FROM retain_history_table;
+                """,
+                """
+                > CREATE MATERIALIZED VIEW retain_history_mv3 WITH (RETAIN HISTORY FOR '30s') AS
+                    SELECT * FROM retain_history_mv2;
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > SELECT * FROM retain_history_mv1;
```

Comment:
Maybe I'm missing some context, but do we actually check that we can read the retained history and that it is correct? I assumed to see some select with `AS OF` here.

Reply:
I figured out that it is not straight-forward to test this with a platform check due to the limitations of testdrive (no possible to use variables, value for `AS OF` needs to be a constant). This is why I introduced `test/retain-history/mzcompose` where I perform the "real" testing.

(Though, I am open for ideas.)

Reply:
Not sure if this will help you, but testdrive supports the following syntax:

```
$ set-from-sql var=source-size
SELECT size FROM mz_sources WHERE name = 'gen'
```

Given that you already have the retain-history mzcompose , I think there is no need to try to combine the two. My only request would be to introduce one query in the platform check that returns a `contains: is not valid for all inputs` error, if it is possible to easily construct such a query that would be reliable.

Reply:
✅ I significantly extended the platform check and found a way to use `set-from-sql`.